src/ion/data_structures.rs

```diff
@@ -396,6 +396,70 @@ impl BlockparamIn {
     }
 }
 
+impl LiveRanges {
+    pub fn add(&mut self, range: CodeRange) -> LiveRangeIndex {
+        let idx = self.storage.len();
+
+        self.storage.push(LiveRange {
+            range,
+            vreg: VRegIndex::invalid(),
+            bundle: LiveBundleIndex::invalid(),
+            uses_spill_weight_and_flags: 0,
+
+            uses: smallvec![],
+        });
+
+        LiveRangeIndex::new(idx)
```

Comment:
I wonder if we could make these `add` methods a little cleaner by generating a `push` method that returns the newtype'd index to the macro body, and using that? (So the body here would be just `self.push(LiveRange { ... })`)

Reply:
I like that idea! The fewer places the index value is directly manipulated, the better.